src/components/sidebar/routePathView/RoutePathView.tsx

```diff
@@ -43,37 +47,68 @@ class RoutePathView extends React.Component<IRoutePathViewProps, IRoutePathViewS
             decodeURIComponent(navigator.getQueryParam(QueryParams.startTime)));
         const routePath =
             await RoutePathService.fetchRoutePath(routeId, startTime, direction);
-        this.setState({ routePath });
+        this.setState({
+            routePath,
+            isLoading: false,
+        });
     }
 
     public toggleEditing = () => {
         const isEditingDisabled = !this.state.isEditingDisabled;
         this.setState({ isEditingDisabled });
     }
 
-    public save = () => {
-        this.setState({ hasModifications: false });
+    public save = async () => {
+        this.setState({ isLoading: true });
+        try {
+            await RoutePathService.saveRoutePath(this.state.routePath!);
+            this.setState({ hasModifications: false });
+            NotificationStore.addNotification({
+                message: 'Tallennus onnistui',
+                type: NotificationType.SUCCESS,
+            });
+        } catch {
+            NotificationStore.addNotification({
+                message: 'Tallennus epäonnistui',
+                type: NotificationType.ERROR,
+            });
+        }
+        this.setState({ isLoading: false });
+    }
+
+    public onChange = (property: string, value: any) => {
+        this.setState({
+            routePath: { ...this.state.routePath!, [property]: value },
+            hasModifications: true,
```

Comment:
I think to see if we have notifications or not, we should have an initialState saved somewhere and current that we compare to find out if we have modifications

"What if user changes a field from xxx to xx, then back from xx to xxx"

Reply:
Maybe not required for a MVP, but added task so we remember it. https://github.com/HSLdevcom/jore-map-ui/issues/357